tests/v2/validation/rbac/snapshotrbac/snapshot_rbac_test.go

```diff
@@ -76,34 +88,62 @@ func (etcd *SnapshotRBACTestSuite) TestRKE2K3SSnapshotRBAC() {
 	}
 	for _, tt := range tests {
 		if !(strings.Contains(etcd.cluster.ID, "c-m-")) {
-			etcd.T().Skip("Skipping tests since cluster is not of type - k3s or RKE2")
-		}
-		etcd.Run("Set up User with Role "+tt.name, func() {
-			clusterUser, clusterClient, err := rbac.SetupUser(etcd.client, tt.member)
-			require.NoError(etcd.T(), err)
-
-			adminProject, err := etcd.client.Management.Project.Create(projects.NewProjectConfig(etcd.cluster.ID))
-			require.NoError(etcd.T(), err)
-
-			if tt.member == rbac.StandardUser.String() {
-				if strings.Contains(tt.role, "project") {
-					err := users.AddProjectMember(etcd.client, adminProject, clusterUser, tt.role, nil)
-					require.NoError(etcd.T(), err)
-				} else {
-					err := users.AddClusterRoleToUser(etcd.client, etcd.cluster, clusterUser, tt.role, nil)
-					require.NoError(etcd.T(), err)
+			etcd.Run("Set up User with Role "+tt.name, func() {
+				clusterUser, clusterClient, err := rbac.SetupUser(etcd.client, tt.member)
+				require.NoError(etcd.T(), err)
+
+				adminProject, err := etcd.client.Management.Project.Create(projects.NewProjectConfig(etcd.cluster.ID))
+				require.NoError(etcd.T(), err)
+
+				if tt.member == rbac.StandardUser.String() {
+					if strings.Contains(tt.role, "project") {
+						err := users.AddProjectMember(etcd.client, adminProject, clusterUser, tt.role, nil)
+						require.NoError(etcd.T(), err)
+					} else {
+						err := users.AddClusterRoleToUser(etcd.client, etcd.cluster, clusterUser, tt.role, nil)
+						require.NoError(etcd.T(), err)
+					}
 				}
-			}
 
-			relogin, err := clusterClient.ReLogin()
-			require.NoError(etcd.T(), err)
-			clusterClient = relogin
+				relogin, err := clusterClient.ReLogin()
+				require.NoError(etcd.T(), err)
+				clusterClient = relogin
```

Comment:
can we take this out of the loop `if !(strings.Contains(etcd.cluster.ID, "c-m-"))` and have it common for both cluster types? because the only thing that changes is the cluster type.

Reply:
Yes that sounds good I'll move that stuff out.

Reply:
Agree with what Anu is suggesting. We can only have this in the if statement and rest of the code can be moved outside of it 
```
if strings.Contains(etcd.cluster.ID, "c-m-") {
	etcd.testRKE2K3SSnapshotRBAC(tt.role, clusterClient)
	} else {
	etcd.testRKE1SnapshotRBAC(tt.role, clusterClient)
}
```